
 This patch resolves bug #908 where a tape is not properly mounted
 (recognized) during a poll.

 Apply to version 2.2.4 (and perhaps older 2.2.x versions) with
  
 cd <bacula-source>
 patch -p0 <2.2.4-poll-mount.patch
 ./configure (your options)
 make
 ...
 make install

Index: src/stored/dev.c
===================================================================
--- src/stored/dev.c	(revision 5553)
+++ src/stored/dev.c	(working copy)
@@ -1844,7 +1844,8 @@
 
    /* Clean up device packet so it can be reused */
    clear_opened();
-   state &= ~(ST_LABEL|ST_READ|ST_APPEND|ST_EOT|ST_WEOT|ST_EOF);
+   state &= ~(ST_LABEL|ST_READ|ST_APPEND|ST_EOT|ST_WEOT|ST_EOF|
+              ST_MOUNTED|ST_MEDIA|ST_SHORT|ST_FREESPACE_OK|ST_PART_SPOOLED);
    label_type = B_BACULA_LABEL;
    file = block_num = 0;
    file_size = 0;
